modules/calib3d/src/stereosgbm.cpp

```diff
@@ -1175,7 +1175,7 @@ void cv::validateDisparity( InputOutputArray _disp, InputArray _cost, int minDis
                 int d = dptr[x], c = cptr[x];
                 int x2 = x - ((d + DISP_SCALE/2) >> DISP_SHIFT);
 
-                if( disp2cost[x2] < c )
+                if( x2>=0 && x2<cols && disp2cost[x2] < c )
```

Comment:
Shouldn't it be `disp2cost[x2] > c` here?
Please, use same spacing rule as in second expression part: `x2 >= 0 && x2 < cols ...`

Isn't it better to use:
```.cpp
if( d == INVALID_DISP_SCALED )
    continue;
```
like on line 1192?